src/gtk/toga_gtk/widgets/numberinput.py

```diff
@@ -1,41 +1,54 @@
 from gi.repository import Gtk
+import sys
+from travertino.constants import LEFT, RIGHT
 
 from .base import Widget
 
 
 class NumberInput(Widget):
     def create(self):
-        adjustment = Gtk.Adjustment(0, self.interface.min_value,
-                                    self.interface.max_value,
-                                    self.interface.step, 10, 0)
-
+        self.interface.adjustment = Gtk.Adjustment(0, -sys.maxsize, sys.maxsize,
+                                                    1, 10, 0)
         self.native = Gtk.SpinButton()
-        self.native.set_adjustment(adjustment)
         self.native.set_numeric(True)
         self.native.interface = self.interface
 
+        self.native.set_adjustment(self.interface.adjustment)
         self.rehint()
 
     def set_readonly(self, value):
-        self.native.editable = not value
+        self.native.set_property('editable', not value)
 
     def set_step(self, step):
-        self.interface.factory.not_implemented('NumberInput.set_step()')
+        if step:
+            self.native.set_increments(step,step)
 
     def set_min_value(self, value):
-        self.interface.factory.not_implemented('NumberInput.set_min_value()')
+        if value:
+            self.interface.adjustment.set_lower(self.interface.min_value)
 
     def set_max_value(self, value):
-        self.interface.factory.not_implemented('NumberInput.set_max_value()')
+        if value:
+            self.interface.adjustment.set_upper(self.interface.max_value)
 
     def set_value(self, value):
         self.native.set_value(value)
 
     def set_alignment(self, value):
-        self.interface.factory.not_implemented('NumberInput.set_alignment()')
+        if value == LEFT:
+            self.native.set_alignment(0)
+        elif value == RIGHT:
+            self.native.set_alignment(1)
+        else:
+            try:
+                float(value) >=0.0 and float(value) <=1.0
```

Comment:
I thought I caught this last time - but what is this statement doing? It's got odd spacing; it's a non-assigned value, but it's a complex comparison clause. I can't work out what it's purpose is. Also, there shouldn't be a need for type or value checking at the GTK level - the values are already type checked by the interface layer. The role of the implementation layer is to handle the values as provided.

Reply:
I think I need some clarification regarding this - when I do `self.native.set_alignment(0.7)` at the implementation level I am able to get the following result - 

![Output](https://ibin.co/3rY3JqLgJz2C.png)

But I cannot pass the value as `0.7` from the interface - It gives a ValueError
```
 File "/home/yugesh/Documents/beeware/venv/lib/python3.5/site-packages/travertino/declaration.py", line 181, in setter
    value, name, choices
ValueError: Invalid value '0.7' for property 'alignment'; Valid values are: bottom, center, left, right, top
```

Reply:
So, Gtk API supports float values for alignments but our core does not? What if a use case requires the alignment to change based on some calculations that happen while the app is running?

Reply:
That may be a use case that *GTK* supports, but that doesn't mean *Toga* supports it. 

If you can make a particularly compelling reason for why a fractional alignment is useful or valuable, then the fix is to add that support at the interface level, and then require *all* backends to support it. In this case, I can't see any particularly good reason why it's useful.